From 79e8b56b07091949638211a308dbfbb073539ebb Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 18 Feb 2015 11:07:17 +0000
Subject: [PATCH 15/27] Skip posix-to-windows conversion when '::' is seen

The substring '::' most likely denotes an IPv6 address, not a path.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/msys2_path_conv.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/winsup/cygwin/msys2_path_conv.cc b/winsup/cygwin/msys2_path_conv.cc
index 74ae36b..f032ddf 100644
--- a/winsup/cygwin/msys2_path_conv.cc
+++ b/winsup/cygwin/msys2_path_conv.cc
@@ -362,6 +362,11 @@ skip_p2w:
             if (it + 1 < end && it[1] == '~')
                 goto skip_p2w;
             break;
+        case ':':
+            // Avoid mangling IPv6 addresses
+            if (it + 1 < end && it[1] == ':')
+                goto skip_p2w;
+            break;
         ++it;
     }
     it = *src;
-- 
2.6.3

